### PR TITLE
Use travis deploy config

### PR DESCRIPTION
**Pull request by kevmoo**
Fixes https://github.com/dart-lang/angular/issues/978

### CODE REVIEW

File: tool/travis.sh [hunk 1]
```diff
@@ -56,8 +56,8 @@ while (( "$#" )); do
     ;;
   command_7) echo
     echo -e '\033[1mTASK: command_7\033[22m'
-    echo -e 'pub run build_runner build --config=release --fail-on-severe'
-    pub run build_runner build --config=release --fail-on-severe || EXIT_CODE=$?
+    echo -e 'pub run build_runner build --config=release --fail-on-severe -o build'
+    pub run build_runner build --config=release --fail-on-severe -o build || EXIT_CODE=$?
```
Comment (kevmoo) [inline tool/travis.sh]: This was in the hacker news mono_repo file, but not synced here!

File: .travis.yml [hunk 2]
```diff
@@ -150,9 +150,19 @@ branches:
   only: [master]
 
 # MANUALLY ADDED. Do not remove.
-after_success:
-  - ./tool/deploy.sh
-# END MANUAL EDITS.
+deploy:
```
Comment (kevmoo) [inline .travis.yml]: Actually does the work to install the tools – and only when we want them installed!

File: .travis.yml [hunk 3]
```diff
@@ -150,9 +150,19 @@ branches:
...
+  skip_cleanup: true
+  on:
+    repo: dart-lang/angular
+    #branch: master
+    branch: i978_better_deploy
+    condition: -d "examples/hacker_news_pwa/build/web"
+  provider: firebase
+  token:
+    # Created by kevmoo
+    secure: "onLskcM902pl6N8gtTp/oqRQmtLt6xjnJx8WW8r0tIaYdiQyhHn3sm6rFFyEiYBuIjpNS1+zK3pkDZ2dT1ZIVXPeQ7Zyo1TkbwqRDdWIuy9Qcv8RjAnRvhC2jh41e5nFuLQ1cNJQV6JtpF+sXfc7vpFs5Iaq+X9M4E7Z/HqHQ8IDRXXTLMvZpshSHr+ReaLzNp5p9ShxNNGEoDERZ4+O5sd39Al7fEoczSQNsq9BObWyhmR04lKBGLf7ECasp/qbrWExkiRODS1Izwd1LfLtHeNUoTHfbRs3tSxB74eVp+BBwtkni5Zp+llY/DFZgxw2GWC7iVT0PbjAqzdb+DKgiYjdIVC8wwOMV4C3eeZbwVGiAAQtt+NgpLKKpslp/+Cj+F8sDgSWeUJmtxQhNtV/sJDD+Ai4ijOubjba0WuGmuDgNzVWMPrsKNL+e6sMATR1omAWfUKIaYjAo0vk2KFoQbrwpl8qRQLrchrzgTi6JEs000cnVJYimlmcSK8dJdd1wkw/wiOAoD9oR9/ApUQaC7p8GnqIjhwR+dI5efj/zteBZI3HL/EbVEarpRlx4GKcCzYmC7ANSCJtex/d6fArBnWSKvoQHcedLkAbqpB622ir+d+QeRRSbyxSYkVgKFDNje8j32wZ3zNYXPqzwEPFy/ecpDTkrY9ltbZwncnsrtc="
```
Comment (kevmoo) [inline .travis.yml]: This is what's documented: https://docs.travis-ci.com/user/deployment/firebase/#Generating-your-Firebase-token

Wonder if it's necessary if have the key in settings...

Comment (matanlurey) [reply .travis.yml]: Why can't we go with the `$FIREBASE_TOKEN` approach? Seems easier.

**Comment by kevmoo** [comment]
Okay. I'll go through another test round, though...

**Comment by kevmoo** [comment]
Seems to work - https://travis-ci.org/dart-lang/angular/jobs/347946765#L1360

Updating PR now...

**Comment by kevmoo** [comment]
@matanlurey – PTAL

**Comment by matanlurey** [comment]
I'd like to hold on this for a little bit. I'm changing the test configurations for travis, but I'm happy to merge in your changes related to firebase `deploy: ` after.

**Comment by kevmoo** [comment]
K. I'll leave it...

**Comment by matanlurey** [comment]
Closed by work in #987.

**Pull request closed without merging**